### PR TITLE
Check against artifacts without signatures

### PR DESCRIPTION
**Pull request by pschoenfelder**
Adding an option to require that artifacts have signatures.

### CODE REVIEW

File: SingularityExecutor/src/main/java/com/hubspot/singularity/executor/config/SingularityExecutorConfiguration.java [hunk 1]
```diff
@@ -188,6 +188,11 @@
   @JsonProperty
   private boolean failTaskOnInvalidArtifactSignature = true;
 
+  @JsonProperty
+  private boolean failOnArtifactWithNoMatchingSignature = false;
+
+  private boolean failOnSignatureWithNoMatchingArtifact = false;
```
Comment (rosalind210) [inline SingularityExecutor/src/main/java/com/hubspot/singularity/executor/config/SingularityExecutorConfiguration.java:194]: Why doesn't this field need the `@JsonProperty` annotation?

File: SingularityExecutor/src/main/java/com/hubspot/singularity/executor/config/SingularityExecutorConfiguration.java [hunk 2]
```diff
@@ -602,6 +605,16 @@ public void setFailTaskOnInvalidArtifactSignature(
     this.failTaskOnInvalidArtifactSignature = failTaskOnInvalidArtifactSignature;
   }
 
+  public boolean isFailOnArtifactWithNoMatchingSignature() {
+    return failOnArtifactWithNoMatchingSignature;
+  }
+
+  public void seFailOnArtifactWithNoMatchingSignature(
```
Comment (rosalind210) [inline SingularityExecutor/src/main/java/com/hubspot/singularity/executor/config/SingularityExecutorConfiguration.java:612]: I think you meant `setFailOnArtifactWithNoMatchingSignature`

File: SingularityExecutor/src/main/java/com/hubspot/singularity/executor/config/SingularityExecutorConfiguration.java (hunk 1 above)
Comment (pschoenfelder) [reply SingularityExecutor/src/main/java/com/hubspot/singularity/executor/config/SingularityExecutorConfiguration.java:194]: Not sure, just moved that one up here to be with the others.

**Comment by tpetr** [comment]
:ship:

**Pull request merged**